### PR TITLE
Fix lifetime issue for functions

### PR DESCRIPTION
**Pull request by xuebinsu**
Previously, to avoid double creation error, we maintianed a global scope
of functions and throwed an error if the name of a new function
conflicts with those in the global scope. However, this makes it very
inconvenient if the user want to change the function definition, because
they will need to restart the process, i.e. the IPython kernel if they
are using Jupyter Notebook in that case.

This patch removes the global scope so that the user can make the change
by re-creating the function object. To avoid double creation error, we
force all UDFs to be TEMP objects so that

> An object is alive in database system if and only if both the Python
> object and the db connection are alive.

This is the rule of reasoning the livetime of an object, including
tables, functions, operators, and types, unless the object is saved
explicitly by the user.

### CODE REVIEW

File: greenplumpython/func.py [hunk 1]
```diff
@@ -206,27 +206,19 @@ def __init__(
...
         qualified_name = (
-            (name if schema is None else f"{schema}.{name}")
-            if name is not None
-            else f"pg_temp.{wrapped_func.__name__}"
+            f"pg_temp.{_name}"
             if wrapped_func is not None
-            else None
+            else _name
+            if schema is None
+            else f"{schema}.{_name}"
```
Comment (beeender) [inline greenplumpython/func.py:219]: Does this mean:
```
(f"pg_temp.{_name}") if wrapped_func is not None
else (_name if schema is None else f"{schema}.{_name}")
```

File: greenplumpython/func.py [hunk 2]
```diff
@@ -86,7 +86,21 @@ def apply(self, expand: Optional[bool] = None, as_name: Optional[str] = None) ->
         if expand and as_name is None:
             as_name = "func_" + uuid4().hex
         if not expand and as_name is None:
-            as_name = self.function.unwrap().__name__
+            if isinstance(self.function, AggregateFunction):
+                try:
+                    self.function.transition_function.unwrap()
+                except:
```
[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/func.py:92]: Transition_function does not contain wrap_func for already existing function

[the remark this answers is not in the document]
Comment (Quasimewdo) [reply greenplumpython/func.py:92]: Because for python udf, the `qualified_name` will be too complex, and we still need to test i`f wrapped_func is None
```
_name = "func_" + uuid4().hex if wrapped_func is not None else name
qualified_name = (
            f"pg_temp.{_name}"
            if wrapped_func is not None
            else _name
            if schema is None
            else f"{schema}.{_name}"
        )
```

**Review by beeender — APPROVED** [review]

**Review by Quasimewdo — APPROVED** [review]
LGTM
